modules/terraform/output.go

```diff
@@ -131,16 +135,71 @@ func OutputMapE(t *testing.T, options *Options, key string) (map[string]string,
 	return resultMap, nil
 }
 
-// OutputForKeysE calls terraform output for the given key list and returns values as a map.
+// OutputForKeys calls terraform output for the given key list and returns values as a map.
 // If keys not found in the output, fails the test
 func OutputForKeys(t *testing.T, options *Options, keys []string) map[string]interface{} {
 	out, err := OutputForKeysE(t, options, keys)
 	if err != nil {
 		t.Fatal(err)
 	}
+
 	return out
 }
 
+// OutputForKeysTgE calls terragrunt output-all for the given key list and returns values as a map.
+// If keys not found in the output, fails the test
+func OutputForKeysTgE(t *testing.T, options *Options, keys []string) (map[string]interface{}, error) {
```

Comment:
Having read through this again, I am not sure this is the right interface. This doesn't appear to handle the case where you can have multiple json objects nested in the outputs.

You probably need:

- A base function that parses out all the jsons as a list. Bonus points if you can map it back to the original modules, although that is probably not possible due to the interleaving.
- This function should then use the base function to get all the json outputs from all the modules, and then search for the keys in each json output. The map will also need to be adjusted to be a list: you can have multiple outputs for the given `key` if it appears in multiple jsons.

Given the scope and complexity, this is probably better suited in another PR. I think it best to remove this function altogether for now, and file an issue specifying support for `output-all` with this comment pasted in.